src/test/java/com/google/api/codegen/metacode/SampleInitCodeTest.java

```diff
@@ -215,57 +229,63 @@ public void testEmbeddedFieldList() throws Exception {
   }
 
   @Test(expected = IllegalArgumentException.class)
+  public void testFormattedFieldBadField() throws Exception {
+    String fieldSpec = "name%entity=test";
```

Comment:
The changes to the conditionals look good, but I think there is still an unhandled error case. This test should still throw an exception if you remove the "=test" component, but at the moment I don't think it will.